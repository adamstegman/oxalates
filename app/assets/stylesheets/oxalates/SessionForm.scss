@import "./actions-form";
@import "./constants";

.item {
  min-width: $session_width;
  max-width: $session_width;
}

form.session {
  @extend %actions-form;
  @extend .item;
}
