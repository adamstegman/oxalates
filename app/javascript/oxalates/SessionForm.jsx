import React from 'react';
import PropTypes from 'prop-types';

export const SessionForm = ({ error, password, onChangePassword, onStopAuthentication }) => {
  const _onStopAuthentication = event => {
    event.preventDefault();
    onStopAuthentication(password);
  };
  const _onChangePassword = event => onChangePassword(event.target.value);

  let errorMessage = null;
  if (error) {
    errorMessage = <p className="item">{error}</p>;
  }

  return (
    <form className="session" onSubmit={_onStopAuthentication}>
      <label htmlFor="password">Password</label>
      <input type="password"
             name="password"
             placeholder="Password"
             autoFocus
             value={password}
             onChange={_onChangePassword} />
      <input type="submit" value="Log in" />
      {errorMessage}
    </form>
  );
};

SessionForm.propTypes = {
  error: PropTypes.string,
  password: PropTypes.string,
  onChangePassword: PropTypes.func,
  onStopAuthentication: PropTypes.func,
};
