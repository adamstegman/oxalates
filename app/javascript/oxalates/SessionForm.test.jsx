import React from 'react';
import renderer from 'react-test-renderer';

import { SessionForm } from './SessionForm';

const password = 'password';

test('SessionForm renders the authentication form', () => {
  const component = renderer.create(
    <SessionForm error={null}
                 password={password}
                 onChangePassword={() => {}}
                 onStopAuthentication={() => {}} />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('SessionForm renders the failed authentication form', () => {
  const error = 'some error';
  const component = renderer.create(
    <SessionForm error={error}
                 password={password}
                 onChangePassword={() => {}}
                 onStopAuthentication={() => {}} />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
