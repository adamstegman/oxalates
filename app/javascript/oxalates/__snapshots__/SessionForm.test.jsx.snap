// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SessionForm renders the authentication form 1`] = `
<form
  className="session"
  onSubmit={[Function]}
>
  <label
    htmlFor="password"
  >
    Password
  </label>
  <input
    autoFocus={true}
    name="password"
    onChange={[Function]}
    placeholder="Password"
    type="password"
    value="password"
  />
  <input
    type="submit"
    value="Log in"
  />
</form>
`;

exports[`SessionForm renders the failed authentication form 1`] = `
<form
  className="session"
  onSubmit={[Function]}
>
  <label
    htmlFor="password"
  >
    Password
  </label>
  <input
    autoFocus={true}
    name="password"
    onChange={[Function]}
    placeholder="Password"
    type="password"
    value="password"
  />
  <input
    type="submit"
    value="Log in"
  />
  <p
    className="item"
  >
    some error
  </p>
</form>
`;
